lib/ZumoOta/src/FileManager.cpp

```diff
@@ -0,0 +1,205 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  FileManager realization
+ * 
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "FileManager.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+FileManager::FileManager() {
+    
+}
+    
+
+FileManager::~FileManager() {
+}
+   
+
+bool FileManager::createUploadPage() {

```

Comment:
The file is not to be created by code on the fly, Then you don't need to put it into the FS. You have it as string already.

Figure out how to upload a filesystem in similar way as hex file.  Code must handle a missing file system as an error during startup.

Reply:
FileManager is only to send the webpages to the client. What you describe here will be tacle with further developments in the Uploader.

Reply:
The entire FileManager can be deleted based on discussion in sync meeting from 09.11.2023. Webserver can directly deliver content from file system. No need for own code to load it.

Reply:
I will use it just to initialize the FileSystem.